api/base_controller.go

```diff
@@ -187,34 +189,61 @@ func (c *BaseController) GetSingleObject(r *web.Request) (*web.Response, error)
 // ListObjects handles the fetching of all objects
 func (c *BaseController) ListObjects(r *web.Request) (*web.Response, error) {
 	ctx := r.Context()
-	if ctx.Value("limit").(int) == 0 {
-		log.C(ctx).Debugf("Getting count of %s since max_items is 0", c.objectType)
-		count, err := c.repository.Count(ctx, c.objectType, query.CriteriaForContext(ctx)...)
-		if err != nil {
-			return nil, util.HandleStorageError(err, string(c.objectType))
-		}
+	userCriteria := query.UserCriteriaForContext(ctx)
+	count, err := c.repository.Count(ctx, c.objectType, userCriteria...)
+	if err != nil {
+		return nil, util.HandleStorageError(err, string(c.objectType))
+	}
+
+	limit := web.PageLimitFromContext(ctx)
+	if limit == 0 {
```

Comment:
I thought limited is always set and if not provided default value is used. Limit being zero does not seem possible. If this check is to handle a case in which the PagingFilter's FilterMatchers does not include an entry for a resource that uses base_controller.go, it won't work as line 198 will panic

Reply:
By the spec limit 0 means that the client is interested in the total count of the items only.